### PR TITLE
Fixes #37519 - Host form - Libvirt improvements

### PR DESCRIPTION
**Pull request by stejskalleos**
- Provisioning a host with an image that does not exist has the correct error.
- Fixed error for the Libvirt capacity storage.
- Added error to the field when the image is not found

### CODE REVIEW

File: webpack/assets/javascripts/compute_resource/libvirt.js [hunk 1]
```diff
@@ -54,26 +56,38 @@ export function imageSelected(item) {
 
   if (template) {
     const url = $(item).attr('data-url');
+    // For some reason there are two help blocks
+    // so we need to select the correct one
+    const help = $('#image_selection .form-group > div > .help-block');
```
Comment (MariaAga) [inline webpack/assets/javascripts/compute_resource/libvirt.js:62]: In which situation are there two?

File: webpack/assets/javascripts/compute_resource/libvirt.js [hunk 2]
```diff
@@ -54,26 +56,38 @@ export function imageSelected(item) {
...
-          .children('.fields')
-          .find('[id$=format_type]')[0].value = 'qcow2';
+        $('#storage_volumes .fields').find('[id$=format_type]')[0].value =
+          'qcow2';
+      },
+      error() {
+        help.html(
+          $('<span />')
+            .addClass('text-danger')
+            .text('Image not found')
```
Comment (MariaAga) [inline webpack/assets/javascripts/compute_resource/libvirt.js]: Should be translated

**Review by ekohl — COMMENTED** [review]
I'm tempted to ask to split this up so they show up as individual bug fixes in the changelog.

File: webpack/assets/javascripts/compute_resource/libvirt.js (hunk 1 above)
Comment (stejskalleos) [reply webpack/assets/javascripts/compute_resource/libvirt.js:62]: AFAIK there are always two

Comment (stejskalleos) [reply webpack/assets/javascripts/compute_resource/libvirt.js:62]: I don't know why there are two of them. Somewhere, deep in the helper's helper methods, the help text is generated.

Comment (MariaAga) [reply webpack/assets/javascripts/compute_resource/libvirt.js:62]: Can you inspect both and screen shot them? I only see one without this fix

Comment (stejskalleos) [reply webpack/assets/javascripts/compute_resource/libvirt.js:62]: help-block

For `#image_selection .help-block` selector there are two blocks.

**Comment by MariaAga** [comment]
not sure how to recreate the bug, and the pr needs splitting but the .erb and .js code looks ok

**Comment by stejskalleos** [comment]
Updated and split into two (https://github.com/theforeman/foreman/pull/10232)

**Comment by nofaralfasi** [comment]
> * Fixed error for the Libvirt capacity storage.

@stejskalleos What is exactly fixed here? I'm not sure how to test that.

Update: I see now that `capacity` was undefined before. However, that still doesn't fix the issue. The values are not taken from the form but from somewhere else (capacity is always 128G), so I don't know what's the point of checking that.

**Comment by stejskalleos** [comment]
@nofaralfasi I don't have that problem, when I set 33G size, then it works as expected:
```
qemu-img info /home/lstejska/VirtualMachines/ken-hentges.virtual.lan-disk1
=>
image: /home/lstejska/VirtualMachines/ken-hentges.virtual.lan-disk1
file format: qcow2
virtual size: 33 GiB (35433480192 bytes)
disk size: 196 KiB
cluster_size: 65536
Format specific information:
    compat: 0.10
    compression type: zlib
    refcount bits: 16
Child node '/file':
    filename: /home/lstejska/VirtualMachines/ken-hentges.virtual.lan-disk1
    protocol type: file
    file length: 193 KiB (197632 bytes)
    disk size: 196 KiB


```

**Comment by nofaralfasi** [comment]
> @nofaralfasi I don't have that problem, when I set 33G size, then it works as expected:

That's my point. The result is not affected by the JS code inside the `imageSelected` function. Neither the format_type nor the size (capacity) are impacted. 

How did you test it? Did you use network-based or image-based provisioning? Keep in mind that this function is only triggered during image-based provisioning.

**Comment by nofaralfasi** [comment]
I accidentally closed the PR :)

**Comment by stejskalleos** [comment]
Even when I use imaged-based provisioning (qcow2 image), it works fine for me.

**Comment by ekohl** [comment]
Random thought: could it be thin provisioning?

**Comment by stejskalleos** [comment]
@nofaralfasi any other comments? The JS methods have been fixed, should be working and ready for merge.

**Review by nofaralfasi — APPROVED** [review]
While this isn’t perfect, it fixes the original code that wasn’t working. Could you update the RM issue [#37519](https://projects.theforeman.org/issues/37519) to specify that the PXE_LOADER field tooltip is addressed in a different PR?

**Comment by stejskalleos** [comment]
> While this isn’t perfect

Ouch :rofl: 

The Redmine description updated (same as commit message)

**Comment by nofaralfasi** [comment]
> > While this isn’t perfect
> 
> Ouch 🤣

I mean that you fixed the JS code, and it’s working now, but I don't think it's the best approach. However, since it’s not new code, I think it’s okay to merge it.

**Pull request merged**